Feature/fma 26/owner author switch

zoemaas opened the pull request:
We should merge FMA-25 to make this code review easier

oceanByte committed:
Merge pull request #42 from deltaDAO/development

Development

Commit:
FMA-25 Filter out any non-compliant Gaia-X credentials from the search results in the Federated Catalog

   Added SD compliance filter;

Commit:
FMA-25 Filter out any non-compliant Gaia-X credentials from the search results in the Federated Catalog

   - Refactored of the filter;
   - Added button to UI;

Commit:
FMA-26 Resolve and return human-readable names to users

   - Added author/owner switch at asset list level;
   - Added author/owner switch at asset level;

Commit:
Merge branch 'main' of https://github.com/deltaDAO/portal-rws into feature/FMA-25/query_filter

Commit:
Merge branch 'main' of https://github.com/deltaDAO/portal-rws into feature/FMA-26/owner_author_switch

Commit:
FMA-25 Filter out any non-compliant Gaia-X credentials from the search results in the Federated Catalog

   - Changed the filter button text;

Commit:
FMA-26 Resolve and return human-readable names to users

   - Added ellipsis to the name if greater than 25 chars;

Commit:
Merge branch 'feature/FMA-25/query_filter' of github.com:Sphereon/portal-rws into feature/FMA-26/owner_author_switch

nklomp commented on src/@types/MetaData.d.ts:
```diff
@@ -20,6 +20,7 @@ export interface AdditionalInformationMarket extends AdditionalInformation {
   consent: {
     noPersonalData: boolean
   }
+  isInCompliance: boolean
```
I think this is not a good name. compliant, already means something is complying, and already suggest a binary/boolean value. gxCompliant would make more sense. Or better:

```json
compliance: {
    gx: boolean
}
````

zoemaas replied:
Fixed!

nklomp commented on src/components/organisms/AssetContent/MetaMain.tsx:
```diff
@@ -65,7 +64,9 @@ export default function MetaMain(): ReactElement {
             account={owner}
-            verifiedServiceProviderName={verifiedServiceProviderName}
+            verifiedServiceProviderName={
+              isServiceSelfDescriptionVerified ? author : undefined
```
This now can lead to an undefined, whilst previously the verifiedServiceProviderName was used. Shouldn't that be used instead of undefined?

zoemaas replied:
Fixed!

nklomp commented on src/components/templates/Search/Filters.tsx:
```diff
@@ -23,39 +24,61 @@ const accessFilterItems = [
 ]
 
+const complianceFilterItems = [
+  { display: 'in compliance', value: FilterByIsInComplianceOptions.True }
```
Gaia-X compliant

zoemaas replied:
Fixed!

nklomp commented on src/components/templates/Search/Filters.tsx:
```diff
@@ -23,39 +24,61 @@ const accessFilterItems = [
+          urlLocation = `${urlLocation}&accessType=${filter}`
+        } else if (filterType === 'serviceType') {
+          urlLocation = `${urlLocation}&serviceType=${filter}`
+        } else if (filterType === 'complianceType') {
```
Hmzz, Not sure this is correct. Compliance can both be true or false for services (not sure what accessType is).

Please create a ticket for this, as I highly doubt this is correct (same applies to the code above for the compliance filter parts)

zoemaas replied:
The value of `accessType` can be 'Download' or 'Compute' and it is set during the Dataset (serviceType) publication. Download means public access and Compute means private access, which allows the data to be processed only by selected Algorthms (serviceType). Reference: https://docs.oceanprotocol.com/building-with-ocean/compute-to-data/compute-to-data-datasets-algorithms The filters can be applied together or separately. I'm not sure if I understand what's wrong.

zoemaas replied:
In addition Algorithm (serviceType) is always Compute

nklomp commented on src/utils/metadata.ts:
```diff
@@ -261,7 +261,20 @@ export function getInitialPublishFormDatasetsValues(
-export function transformPublishFormToMetadata(
+const verifySelfDescription = async (args: {
+  url?: string
+  raw?: any
```
Please rename this to sd or selfDescription. It becomes confusing, since raw contains the SD, whilst later a raw param is used as a boolean

zoemaas replied:
Fixed!

nklomp commented on src/utils/web3.ts:
```diff
@@ -31,6 +31,15 @@ export function accountTruncate(account: string): string {
   return truncated
 }
 
+export function nameTruncate(name: string): string {
```
It would be nicer to have an optional maxLength param, and use that when calling this code.

zoemaas replied:
Field added

nklomp reviewed (changes requested):
See remarks

Commit:
FMA-26 Resolve and return human-readable names to users

   - Changed isInCompliance to compliance.gx;

Commit:
FMA-26 Resolve and return human-readable names to users

   - Changed undefined to verifiedServiceProviderName;

Commit:
FMA-26 Resolve and return human-readable names to users

   - Fixed the total amount issue;
   - Added maxLength to nameTruncate function;
   - Changed raw to rawSD in verifySelfDescription function;

maikel-maas commented:
Closing this PR since the code is already in feature/FMA-47/vp-support
